heidrun/getty.rb

```diff
@@ -0,0 +1,134 @@
+LANGUAGE_MAP = {
+  'multiple languages' => 'mul'
+}
+
+DATA_PROVIDER_MAP = {
+  'GETTY_ROSETTA' => 'Getty Research Institute',
+  'GETTY_OCP' => 'Getty Research Institute'
+}
+
+build_url = lambda do |p|
+  source = p.root[Krikri::PrimoParser.record('control', 'sourceid').join('/')]
+           .first.value
+  id = p.root[Krikri::PrimoParser.record('control', 'recordid').join('/')]
+       .first.value
+
+  base_url = 'http://primo.getty.edu/primo_library/libweb/action/dlDisplay.do'
+
+  case source
+  when 'GETTY_ROSETTA'
+    "#{base_url}?vid=GRI&afterPDS=true&institution=01GRI&docId=#{id}"
+  when 'GETTY_OCP'
+    "#{base_url}?vid=GRI-OCP&afterPDS=true&institution=01GRI&docId=#{id}"
+  end
+end
+
+lookup_language = lambda do |i|
```

Comment:
Ah, just me being silly.  The language code will either be an iso639-2b code, or the string 'multiple languages', and in the latter case we just want to emit 'mul'.  I'd used a hash lookup with default to allow us to override multiple special values, but it's not really necessary.  I've switched it to a conditional to make this clearer.